Zend/zend_closures.c

```diff
@@ -553,7 +553,7 @@ static HashTable *zend_closure_get_gc(zval *obj, zval **table, int *n) /* {{{ */
 	zend_closure *closure = (zend_closure *)Z_OBJ_P(obj);
 
 	*table = Z_TYPE(closure->this_ptr) != IS_NULL ? &closure->this_ptr : NULL;
-	*n = Z_TYPE(closure->this_ptr) != IS_NULL ? 1 : 0;
```

Comment:
Please don't drop ternaries in cases where the return type is something other than zend_bool. While it comes down to the same thing due to implicit integer widening in C,  there's a semantic difference between `true` and `1`.

Reply:
> Please don't drop ternaries in cases where the return type is something other than zend_bool

@nikic What are the cases that the return type is `zend_bool`? Should I look into the macro return, for example?

> While it comes down to the same thing due to implicit integer widening in C, there's a semantic difference between `true` and `1`.

I thought that **ever** comparison like `{something} !=,==,> {somethingElse}` would always return `1` or `0`.